man/systemd-nspawn.xml

```diff
@@ -980,6 +980,20 @@
         effect.</para></listitem>
       </varlistentry>
 
+      <varlistentry>
+        <term><varname>--notify-mode=</varname></term>
+
+        <listitem><para>Configures support for notifications from init process in the container.
+        <varname>--notify-mode=</varname> supports three options: <option>no</option>, <option>yes</option>, and <option>ready</option>.
+        Option <option>no</option> do not enable notifications between systemd-nspawn and
+        init process in the container and systemd-nspawn notifies systemd with a <literal>READY=1</literal>
+        message when the init process is created. Option <option>yes</option> enables notifications
+        but do nothing with them, systemd-nspawn notifies systemd with a <literal>READY=1</literal>
+        message when the init process is created. Option <option>ready</option>
+        enables notifications and wait for the <literal>READY=1</literal> message from
+        the init process in the container before sending its own to systemd.</para></listitem>
+      </varlistentry>
```

Comment:
> Do you have any specific behaviours in mind for readiness notifications that you'd like to see added?

There was status events that the "rkt api service" (an unprivileged daemon running on the host) would need to gather service starts and stops from containers (see details in https://github.com/systemd/systemd/issues/1369#issuecomment-147398131). You suggested to use "PROPAGATE_X_FOO=bar".

But I don't know yet if sd_notify is the right mechanism to implement that feature... so I suggest to not add any abstraction for that yet, at least in this PR. We can revisit this idea later when we have a better idea of how that would work for the rkt api service. I agree with the "no premature abstraction" idea.